### ISSUE TITLE
Panic for "non-positive interval for newticker"

### ISSUE THREAD

**Issue by weeco** [jackc/pgx#567]
Using pgx v4.
If I don't set a healthcheck period I'll receive a panic. I believe it makes sense to default to reasonable default duration for the healthcheck period. 

```
panic: non-positive interval for NewTicker

goroutine 55 [running]:
time.NewTicker(0x0, 0x0)
	/usr/local/Cellar/go/1.12.6/libexec/src/time/tick.go:23 +0x1f8
github.com/jackc/pgx/v4/pgxpool.(*Pool).backgroundHealthCheck(0xc0002bd880)
	/Users/xy/go/pkg/mod/github.com/jackc/pgx/v4@v4.0.0-pre1/pgxpool/pool.go:247 +0x4b
created by github.com/jackc/pgx/v4/pgxpool.ConnectConfig
	/Users/xy/go/pkg/mod/github.com/jackc/pgx/v4@v4.0.0-pre1/pgxpool/pool.go:162 +0x27e
```

**Comment by jackc** [comment]
There is a default, but only when the `pgxpool.Config` gets created by `pgxpool.ParseConfig`.

I went this route because as a general rule I'm wary of using the zero value to indicate a default. That technique is problematic when a zero value is a legitimate option. I prefer a constructor that sets defaults and then to override specific fields as needed.  

From the doc for `pgxpool.Config`:

> It is highly recommended to modify a Config returned by ParseConfig rather than to construct a Config from scratch.

I suppose this is a little ambiguous. 

There are a few possibilities to improve this.

1. Obviously, we just use the zero value to indicate the default -- but as I wrote above I'd rather not do that if possible.
2. The documentation for `pgxpool.Config` could be made clearer.
3. An invalid `HealthCheckPeriod` could be checked for and a better error returned if invalid.
4. We could enforce that `Config` was created by `ParseConfig`.

I'm kind of leaning toward 4. Otherwise adding a new config option with an invalid zero value would be a breaking change.
